crates/typst-library/src/visualize/image/svg.rs

```diff
@@ -287,3 +312,82 @@ impl FontResolver<'_> {
         Some(id)
     }
 }
+
+struct ImageResolver<'a> {
+    /// The world we use to check if resolved images in the SVG are within the project root.
+    world: Tracked<'a, dyn World + 'a>,
+    /// The span of the file loading the SVG.
+    span: &'a Span,
+    /// Path to the SVG file or an empty string if the SVG is given as bytes.
+    svg_path: &'a EcoString,
+    /// The first error message when loading a linked image.
+    error_msg: EcoString,
+    /// The linked image causing the error
+    error_href: EcoString,
+}
+
+impl<'a> ImageResolver<'a> {
+    fn new(
+        world: Tracked<'a, dyn World + 'a>,
+        span: &'a Span,
+        svg_path: &'a EcoString,
+    ) -> Self {
+        Self {
+            world,
+            span,
+            svg_path,
+            error_msg: EcoString::new(),
+            error_href: EcoString::new(),
+        }
+    }
+
+    fn load(&mut self, href: &str) -> Option<usvg::ImageKind> {
+        // Don't load more linked images if there's already an error.
+        if !self.error_msg.is_empty() {
+            return None;
+        }
+
+        // Resolve the path to the linked image.
+        let href_path = VirtualPath::new(self.svg_path.as_str()).join(href);
+        let href_full = href_path.as_rooted_path().to_str().unwrap();
+        let href_file_result = self.span.resolve_path(href_full).at(*self.span);
+        if let Err(err) = href_file_result {
+            self.error_msg = eco_format!("TODO {:?}", err);
```

Comment:
Is there a better way to format this error. I'm not sure how to trigger this code path either.